sql/core/src/main/scala/org/apache/spark/sql/execution/joins/HashedRelation.scala

```diff
@@ -164,9 +164,8 @@ private[joins] class UnsafeHashedRelation(
   def getValue(key: InternalRow): InternalRow = {
     val unsafeKey = key.asInstanceOf[UnsafeRow]
     val map = binaryMap  // avoid the compiler error
-    val loc = new map.Location  // this could be allocated in stack
-    binaryMap.safeLookup(unsafeKey.getBaseObject, unsafeKey.getBaseOffset,
-      unsafeKey.getSizeInBytes, loc, unsafeKey.hashCode())
+    val loc = map.lookup(unsafeKey.getBaseObject, unsafeKey.getBaseOffset,
```

Comment:
IIUC, this change makes this part thread-unsafe. Is it OK?

Reply:
this is safe to lookup, and It is different from the get(key: InternalRow).

Reply:
Before this PR, `loc` is allocated at each call of `getValue()`. After this PR, `loc` will be shared within each `binaryMap` that is passed to a constructor of `UnsafeHashedRelation`.
Is this behavior change safe?